### PR TITLE
Refactor globalVars into a Separate Class for Improved Encapsulation

### PR DESCRIPTION
**Pull request by Achal1607**
This PR refactors the `globalVars` implementation to improve code structure and encapsulation:

1. **Moved `globalVars` from `extension.ts` to a Separate File**  
   - Isolated the global variables to a dedicated file.
   
2. **Converted from Namespace to Class with Getters and Setters**  
   - `globalVars` is now a class with getter and setter methods, for better encapsulation and allowing controlled access and modification of global state.

### CODE REVIEW

**Review by sid-srini — DISMISSED** [review]
Looks fine to me. Thanks @Achal1607.
Perhaps @naren2605 may also review whether this is in line with his previous review. Thanks.

File: vscode/src/lsp/listeners/notifications/handlers.ts [hunk 1]
```diff
@@ -74,40 +74,42 @@ const logMessageHandler = (param: any) => {
...
 
 const textEditorDecorationDisposeHandler = (param: any) => {
-    let decorationType = globalVars.decorations.get(param);
+    let decorationType = GlobalState.getInstance().getDecoration(param);
     if (decorationType) {
-        globalVars.decorations.delete(param);
+        GlobalState.getInstance().removeDecoration(param);
         decorationType.dispose();
-        globalVars.decorationParamsByUri.forEach((value, key, map) => {
+        GlobalState.getInstance().getDecorationParamsByUri()
```
[the remark this answers is not in the document]
Comment (Achal1607) [reply vscode/src/lsp/listeners/notifications/handlers.ts]: Thanks for pointing it out. It was unitentional. I have removed it and pushed again.

**Review by sid-srini — DISMISSED** [review]
Thanks 👍

File: vscode/src/globalState.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    private constructor() {
+        this.listeners = new Map<string, string[]>();
+        this.debugPort = -1;
+        this.deactivated = true;
+        this.nbProcessManager = null;
+        this.decorations = new Map<string, TextEditorDecorationType>();
+        this.decorationParamsByUri = new Map<Uri, SetTextEditorDecorationParams>();
+    }
+
+    public static getInstance(): GlobalState {
```
Comment (naren2605) [inline vscode/src/globalState.ts:49]: can we make this as  export const globalState = new GlobalState() and remove getInstance() method

Comment (Achal1607) [reply vscode/src/globalState.ts]: Done. Thanks!

**Comment by naren2605** [comment]
LGTM thanks @Achal1607  for making these changes

**Review by naren2605 — APPROVED** [review]

**Review by sid-srini — APPROVED** [review]
Thanks @Achal1607. LGTM 👍

**Pull request merged**